### ISSUE TITLE
Extend `comparison-with-itself` to support any expressions

### ISSUE THREAD

**Issue by harupy** [astral-sh/ruff#21201]
### Question

Related to https://github.com/pylint-dev/pylint/issues/10713. `comparison-with-itself` only flags comparisons with `Name` or `Name` with certain function calls (e.g., `id(a) == id(a)`). I wonder if it's possible to extend this r
rule to flag any expressions.

```python
# flagged
a == a

# not flagged
a.b == a.b
a.c() == a.c()
```

### Version

0.9.7

**Comment by dylwil3** [comment]
One thing to be careful with is side effects, eg `next(it) == next(it)`. There is some support for detecting possible side effects in the code base so maybe it's still possible.

**Comment by harupy** [comment]
@dylwil3 Thanks for the comment. Good point. Yeah, you might do `a.b == a.b` in tests to ensure the `b` property result doesn't change :)
